use XString rather than B when possible

haarg opened the pull request.

toddr commented on lib/Sub/Quote.pm:
```diff
@@ -37,6 +35,41 @@ BEGIN {
 
+BEGIN {
+  local $@;
+  if (eval { require XString; XString->VERSION(0.003) }) {
```
We want it to be 0.005 to support 5.8

haarg replied:
That shouldn't matter. A user on 5.8 won't be able to install 0.003, so they won't ever be able to get to this check.

haarg replied:
Or more clearly, a 5.8 user would need to have installed 0.005 for the require to succeed, so the version check would already be passing.

haarg committed:
use XString::perlstring when possible rather than B::perlstring

B has a rather large memort footprint, and we don't need most of what it
provides. We only need the perlstring function.

Refactor our conditionals to just have a _perlstring function, setting
it up at compile time. For 5.10.0, just always use our reimplementation
to simplify things. Add some comments to explain the logic.

haarg committed:
refactor altered quotify tests

Move quotify tests for alternate perl versions out of t, since they
aren't relevant to normal users.

Refactor setup in quotify tests to make it easier to initialize
different environments.

toddr commented:
looks like it needs a rebase but is otherwise fine?
